litestar/response/base.py

```diff
@@ -401,6 +402,8 @@ def to_asgi_response(
             An ASGIResponse instance.
         """
 
+        warn_deprecation("2.1", "app", "parameter", removal_in="3.0.0")
```

Comment:
Please use keyword arguments here for everything. That makes it more readable! (=

Also, this warning should only be raised when the app parameter is actually used, so there's 3 things left to do here:

1. Change the parameter to be `app: Litestar | None`
2. Place the warning inside an `if app is not None` block
3. Remove the usage of the `app` parameter throughout the codebase

This applies to all the other instances as well.

Reply:
Ah thanks!

Reply:
Ok I pushed up the changes and thanks for the guidance!

One thing to note is that [here](https://github.com/wevonosky/litestar/blob/main/litestar/handlers/http_handlers/_utils.py#L65) and [here](https://github.com/wevonosky/litestar/blob/main/litestar/handlers/http_handlers/_utils.py#L151) the app parameter is unused. I am realizing as I am typing this that the type into those two `handler` methods should probably also include `None`

Reply:
Oof failing tests, working on it